INSTALL.md

```diff
@@ -2,34 +2,99 @@
 
 Installation instructions based on: `https://www.gamingonlinux.com/articles/first-steps-with-openvr-and-the-vive-on-linux.7229`
 
+## Install Ubuntu 16.04 LTS.
+## Install latest graphics card driver.
+E.g. for `GeForce GTX 970/PCIe/SSE2` use driver version `384.69` or later. You
+can find the latest Nvidia drivers on
+https://launchpad.net/~graphics-drivers/+archive/ubuntu/ppa .
 
-## Download and build Christoph's OpenVR SDK fork:
+## Copy udev rules.
+Copy the file `88-vive.rules` to the folder `/etc/udev/rules.d`. Then run:
+```
+sudo /etc/init.d/udev restart
+```
 
-      cd ~
-      mkdir libraries
-      cd libraries
-      git clone https://github.com/ChristophHaag/openvr.git
-      cd openvr
-      mkdir build
-      cd build
-      cmake -DCMAKE_BUILD_TYPE=Release ../
-      make
+## Connect and setup steamvr hardware to your computer.
 
-## Allow hardware access
-Then plug-in VIVE to your computer and make sure you can see the devices on `/dev/hidraw[1-6]`.
+## Install steam via ubuntu software center (if the debian package not works).
+After start, steam should detect connected vr hardware and ask to install
+required driver. Agree, to install steamvr. 
 
-Copy the file `88-vive.rules` to the folder `/etc/udev/rules.d`. Then run:
+## Start steam and steamvr via vr icon in steam.
+Steam will probably show an error messages that some i386 shared libraries are
+missing. Try to install all those shared libraries in the `*:i386` version.
+Afterwards, do the `Room setup` via steamvr. Close steam and reboot the
+computer.
 
-      sudo /etc/init.d/udev restart
+## Check installation of steamvr and connected vr hardware.
+You should now see a `$HOME/.local/share/Steam` folder. The program `vrcmd`
+should show a list of connected devices. 
+```
+$HOME/.local/share/Steam/steamapps/common/SteamVR/bin/linux64/vrcmd
+```
+E.g. you have a setup with:
+- 2 Base stations,
+- 1 Head mounted display,
+- 1 HTC vive tracker.
 
-## Install Steam and SteamVR
+The program `vrcmd` should show:
+- HMD - generic\_hmd
+- Tracker - {htc}vr\_tracker\_vive\_1\_0
+- Tracking Reference - lh\_basestation\_vive
+- Tracking Reference - lh\_basestation\_vive
 
-Go to `http://store.steampowered.com/` and download Steam for Linux.
-After successfully installing and running Steam, it should store its files on: `~/.local/share/Steam`
+## Create and build catkin workspace.
+### Catkin version of openvr.
+```
+git clone git@github.com:Intermodalics/openvr.git
+```
+### Catkin package vive\_ros with support for htc vive tracker.
+```
+git clone git@github.com:Intermodalics/vive_ros.git
+git checkout feature/support_htc_vive_tracker
+```
+### Build your workspace and copy shared libraries from steamvr.
+```
+catkin build
+source devel/setup.bash
+roscd vive_ros/scripts
+sh copy_steamvr_shared_libraries.sh
```

Comment:
Note: If you copy the library to devel/lib you have to repeat this each time you would clean and rebuild your workspace. 

Ideally cmake finds the libraries using `find_library(vulkan_LIBRARY vulkan PATHS ... NO_DEFAULT_PATH)` and copies the files to devel-space using `file(COPY ... DESTINATION ...)`. But as this is only for a test, copying it manually or using your shell script should be fine.

Reply:
Good suggestion, maybe we can add it as a separate issue?